core/src/test/scala/unit/kafka/server/KafkaConfigTest.scala

```diff
@@ -1188,7 +1188,7 @@ class KafkaConfigTest {
         case TopicConfig.COMPRESSION_ZSTD_LEVEL_CONFIG =>
           assertDynamic(kafkaConfigProp, "5", () => config.zstdCompressionLevel)
         case TopicConfig.SEGMENT_BYTES_CONFIG =>
-          assertDynamic(kafkaConfigProp, 10000, () => config.logSegmentBytes)
+          assertDynamic(kafkaConfigProp, 1048576, () => config.logSegmentBytes)
```

Comment:
Doesn't have to be the min possible value. A very large value like 2GB will ensure that we don't have to change it in future

Reply:
This test uses produce/consume and checks segment sizes. A large value will not be possible without a larger refactor of the test utils responsible for produce/consume.